src/bucket.js

```diff
@@ -1525,6 +1525,43 @@ class Bucket extends common.ServiceObject {
     );
   }
 
+  /**
+   * Lock a previously-defined retention policy. This will prevent changes to
+   * the policy.
+   *
+   * @example
+   * const storage = require('@google-cloud/storage')();
+   * const bucket = storage.bucket('albums');
+   *
+   * bucket.lock(function(err, apiResponse) {});
```

Comment:
As far as I can tell, and in my own experience, yes. But, I can't confirm definitively.

Reply:
Sorry, I'll rephrase, `storage.bucket('albums')` always returns bucket metadata unless something on the API changes or user does something different.

The issue with lock is it's a one time thing for every bucket. Once locked it can't be unlocked. A requirement is that it has the metageneration for the bucket to pass as a precondition.

Reply:
Woops, sorry for the misunderstanding.

`storage.bucket('albums')` won't actually return any metadata through our API. The user can call, `storage.bucket('albums').getMetadata()` which will update the `metadata` property on that Bucket's instance.

In the code within `lock()`, we don't check for `bucket.metadata.metageneration` and instead just always do a metadata refresh to get the latest value. Is there a better way to handle that?

Reply:
Thanks, could you make metageneration a parameter instead for retentionLock() and getMetadata() before calling this method to pass in the generation? I'd rather this operation not do a metadata get as well. It defeats the purpose of optimistic concurrency in this case.

Reply:
Done!